spec/forms/opponents/domestic_abuse_summary_form_spec.rb

```diff
@@ -0,0 +1,154 @@
+      expect(valid_form).to be true
+    end
+
+    context "when the parameters are incomplete" do
+      let(:params) do
+        {
+          "warning_letter_sent" => "false",
+          "warning_letter_sent_details" => "",
+          "police_notified" => "true",
+          "police_notified_details_true" => "New reasons police not notified details",
+          "bail_conditions_set" => "true",
+          "bail_conditions_set_details" => "New bail conditions set details",
+        }
+      end
+
+      it { expect(valid_form).to be false }
+    end
+
+    context "when radio buttons are empty" do
+      let(:params) do
+        {
+          warning_letter_sent: "",
+          police_notified: "",
+          bail_conditions_set: "",
+        }
+      end
+
+      it "is invalid with appropriate errors" do
+        expect(valid_form).to be false
+        expect(das_form.errors[:warning_letter_sent].join).to eq(I18n.t("warning_letter_sent.blank", scope: i18n_scope))
+        expect(das_form.errors[:police_notified].join).to eq(I18n.t("police_notified.blank", scope: i18n_scope))
+        expect(das_form.errors[:bail_conditions_set].join).to eq(I18n.t("bail_conditions_set.blank", scope: i18n_scope))
+      end
+
+      it "generates errors in the right order" do
+        valid_form
+        expect(das_form.errors.attribute_names).to eq(
+          %i[
+            warning_letter_sent
+            police_notified
+            bail_conditions_set
+          ],
+        )
+      end
+    end
+
+    context "when details are empty even though they should be present" do
+      let(:params) do
+        {
+          warning_letter_sent: "false",
+          police_notified: "true",
+          bail_conditions_set: "true",
+          warning_letter_sent_details: "",
+          police_notified_details_true: "",
+          bail_conditions_set_details: "",
+        }
+      end
+
+      it "is invalid with appropriate errors" do
+        expect(valid_form).to be false
+        expect(das_form.errors[:warning_letter_sent_details].join).to eq(I18n.t("warning_letter_sent_details.blank", scope: i18n_scope))
+        expect(das_form.errors[:police_notified_details_true].join).to eq(I18n.t("activemodel.errors.models.opponent.attributes.police_notified_details_true.blank"))
+        expect(das_form.errors[:bail_conditions_set_details].join).to eq(I18n.t("bail_conditions_set_details.blank", scope: i18n_scope))
+      end
+
+      it "generates errors in the right order" do
+        valid_form
+        expect(das_form.errors.attribute_names).to eq(
+          %i[
+            warning_letter_sent_details
+            police_notified_details_true
+            bail_conditions_set_details
+          ],
+        )
+      end
+    end
+  end
+
+  describe "#save" do
+    subject(:save_form) { das_form.save }
+
+    it "creates a new opponent" do
+      save_form
+      expect(das_form).to have_attributes(police_notified_details_true: "New reasons police not notified details")
```

Comment:
I am not clear what this is supposed to be testing? Would it be clearer to test that it has updated the opponent's police_notified_details?  e.g. expect(opponent.police_notified_details).to eq "New reasons police not notified details". Do we want to include tests for update of any of the other opponent attributes e.g. bail_conditions_set_detail ?